### PR TITLE
Adding scuzz, snap and vnc, removing obsolete spelling/dictionary tools and fax (no telco)

### PR DESCRIPTION
**Pull request by elbing**
Signed-off-by: Álvaro Jurado <elbingmiss@gmail.com>

### CODE REVIEW

**Comment by elbing** [comment]
This leaves pending of being ported:
```
aquarela -> CIFS/Samba server
fax -> Fax receiver/sender fs
grap -> pic(1) preprocessor for drawing graphs (another one)
snap -> save/restore static process images fs
spin -> verification tool for models of concurrent systems (historic, maybe interesting)
tapefs -> tape storing system fs
tcs -> character set translator (classic unix program)
vac -> archival copy of trees for venti (harvey has venti)
vnc -> server/viewer remote frame buffer
```

**Comment by elbing** [comment]
ok?

File: sys/src/cmd/snap/take.c [hunk 1]
```diff
@@ -91,7 +91,7 @@ readsection(int32_t pid, char *sec)
 		return nil;
 
 	tot = 0;
-	hdr = (int)((Data*)0)->data;
+	hdr = (int)(intptr_t)((Data*)0)->data;
```
Comment (gmacd) [inline sys/src/cmd/snap/take.c]: So it all looks good, but I don't understand this - is this not dereferencing a null pointer?  It's late and I'm tired, so maybe I've missed something...  (I realise it was the same in the original code)

Comment (elbing) [reply sys/src/cmd/snap/take.c]: no no, you're in right. I hadn't a better idea for this case. It's a case of assignement of smaller integer than de pointer hdr, I think. Let me review it tomorrow in the machine in what I have the branch and I'll paste the right warning/error. Maybe you could have a better solution, I wasn't sure about this and it's ugly. Thanks!

Comment (gmacd) [reply sys/src/cmd/snap/take.c]: You're not making it any worse, so feel free to merge the PR.

Comment (rminnich) [reply sys/src/cmd/snap/take.c]: you might want to use offsetof
#define offsetof(st, m) ((size_t)&(((st *)0)->m))
You're missing an &

Comment (rminnich) [reply sys/src/cmd/snap/take.c]: I think you should fix this before you commit.

Comment (elbing) [reply sys/src/cmd/snap/take.c]: The problem is not with clang, is with gcc:
```
take.c: In function ‘readsection’:
take.c:94:8: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  hdr = (int)((Data*)0)->data;
```
Not sure why clang eats it.

Comment (elbing) [reply sys/src/cmd/snap/take.c]: Thanks, Ron. It works like a charm:
```
	tot = 0;
	hdr = offsetof(Data, data);
	while((n = read(fd, buf, sizeof buf)) > 0) {
```
libc.h always to the rescue!

File: sys/src/cmd/vnc/compat.c [hunk 2]
```diff
@@ -73,7 +73,11 @@ void
 initcompat(void)
 {
 	rfork(RFREND);
-	privup = privalloc();
+	/* I just don't know how to fix this, because
+	 * in harvey up is Mach *mah()->externup and
+	 * this emulation through the proc seems so weird for me.
+	 */
+	privup = (Proc **)privalloc();
 	kerndate = seconds();
```
Comment (elbing) [inline sys/src/cmd/vnc/compat.c]: suggestions?

**Comment by elbing** [comment]
Sorry for adding vnc, it was in the same branch. I pointed a doubt in `sys/src/cmd/vnc/compat.c`

**Pull request merged**